src/libraries/System.Collections/src/System/Collections/Generic/PriorityQueue.cs

```diff
@@ -320,25 +300,19 @@ public void EnqueueRange(IEnumerable<TElement> elements, TPriority priority)
 
             if (_size == 0)
             {
-                using (IEnumerator<TElement> enumerator = elements.GetEnumerator())
+                int i = 0;
```

Comment:
At some point we could consider presizing if elements is an `ICollection<TElement>`.

Reply:
Done.